src/autoscaler/envelopeprocessor/envelope_processor.go

```diff
@@ -192,13 +192,12 @@ func calcNumReqs(envelopes []*loggregator_v2.Envelope) (numRequestsPerAppIdx map
 }
 
 func isContainerMetricEnvelope(e *loggregator_v2.Envelope) bool {
-	// TODO: Check for all container metrics not only memory quota
 	keys := maps.Keys(e.GetGauge().GetMetrics())
 
 	var matchingKeys []string
 
 	for i := range keys {
-		if keys[i] == "memory_quota" || keys[i] == "memory" {
+		if keys[i] == "memory_quota" || keys[i] == "memory" || keys[i] == "cpu" || keys[i] == "cpu_entitlement" {
```

Comment:
unrelated bugfix: checking here for `cpu` fixes broken scaling in case a policy contains only one scaling rule that is based on `cpu`.